domain/application/state/resource.go

```diff
@@ -69,13 +402,103 @@ func (st *State) OpenUnitResource(
 	return resource.Resource{}, nil
 }
 
-// SetRepositoryResources sets the "polled" resource
-// s for the
+// SetRepositoryResources sets the "polled" resources for the
 // application to the provided values. The current data for this
 // application/resource combination will be overwritten.
+// Returns [apperrors.ApplicationNotFound] if the application id doesn't belong to a valid application.
 func (st *State) SetRepositoryResources(
 	ctx context.Context,
 	config resource.SetRepositoryResourcesArgs,
 ) error {
-	return nil
+	db, err := st.DB()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	// Prepare statement to check that the application exists.
+	getAppNameQuery := `
+SELECT name as &applicationDetails.name 
+FROM application 
+WHERE uuid = $applicationDetails.uuid
+`
+	getAppNameStmt, err := st.Prepare(getAppNameQuery, applicationDetails{})
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	type resourceNames []string
+	// Prepare statement to get impacted resources UUID.
+	fetchUUIDsQuery := `
+SELECT uuid as &resourceIdentity.uuid
+FROM v_application_resource
+WHERE  application_uuid = $resourceIdentity.application_uuid
+AND name IN ($resourceNames[:])
+`
+	fetchUUIDsStmt, err := st.Prepare(fetchUUIDsQuery, resourceIdentity{}, resourceNames{})
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	// Prepare statement to update lastPolled value.
+	type lastPolledResource struct {
+		UUID       string    `db:"uuid"`
+		LastPolled time.Time `db:"last_polled"`
+	}
+	updateLastPolledQuery := `
+UPDATE resource 
+SET last_polled=$lastPolledResource.last_polled
+WHERE uuid = $lastPolledResource.uuid
+`
+	updateLastPolledStmt, err := st.Prepare(updateLastPolledQuery, lastPolledResource{})
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	names := make([]string, 0, len(config.Info))
+	for _, info := range config.Info {
+		names = append(names, info.Name)
+	}
+	var resIdentities []resourceIdentity
+	appDetails := applicationDetails{
+		ApplicationID: config.ApplicationID,
+	}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		// Check application exists.
+		err := tx.Query(ctx, getAppNameStmt, appDetails).Get(&appDetails)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return apperrors.ApplicationNotFound
+		}
+		if err != nil {
+			return errors.Capture(err)
+		}
+
+		// Fetch resources UUID.
+		err = tx.Query(ctx, fetchUUIDsStmt, resourceNames(names), resourceIdentity{ApplicationUUID: config.ApplicationID.String()}).GetAll(&resIdentities)
```

Comment:
I think `resourceIdentity` can be instantiated at the start of this function and used in the Prepare.